troop/views.py

```diff
@@ -233,3 +237,60 @@ def form_invalid(self, form):
         response = super().form_invalid(form)
         response.status_code = 422  # Unprocessable Entity
         return response
+
+
+def csv_participant_export(request, troop_number):
+    troop = user_managed_troop(request.user, troop_number)
+    if not troop:
+        raise PermissionDenied()
```

Comment:
Some views use `self.handle_no_permission()`. Can we unify this?

Reply:
`self.handle_no_permission()` is used only once in the codebase. It has the nice side effect of redirecting to login if the user is not authenticated.

I don't know how we can use this mixin in a simple function (which can't inherit a class I guess)

Reply:
Fixed using the class-based view :-)